zerver/lib/events.py

```diff
@@ -324,6 +324,17 @@ def fetch_initial_state_data(
             "event_queue_longpoll_timeout_seconds"
         ] = settings.EVENT_QUEUE_LONGPOLL_TIMEOUT_SECONDS
 
+        # ! initial state[commit - 5]
+
+        if realm.direct_message_permission_group_id:
+            state["realm_direct_message_permission_group_id"] = realm.direct_message_permission_group_id.id
+        else:
+            state["realm_direct_message_permission_group_id"] = -1
+        if realm.direct_message_initiator_group_id:
+            state["realm_direct_message_initiator_group_id"] = realm.direct_message_initiator_group_id.id
+        else:
+            state["realm_direct_message_initiator_group_id"] = -1
```

Comment:
What's the thinking behind the `-1` values here? I think it'd likely be better to just copy the value. Also `realm.direct_message_permission_group_id.id` isn't valid; you skip the trailing `.id` :)